src/event.ts

```diff
@@ -182,6 +184,28 @@ export interface DuplicateDependency extends ErrorBase<ScriptReference> {
   diagnostic: Diagnostic;
 }
 
+/**
+ * A script depends on another in a package that isn't there.
+ */
+export interface DependencyOnMissingPackageJson
+  extends ErrorBase<ScriptReference> {
+  reason: 'dependency-on-missing-package-json';
+  diagnostic: Diagnostic;
+  /**
+   * This is a better error message than the missing-package-json error,
+   * so if we'd be going to display both, we should only display this one.
+   */
+  supercedes: Failure;
```

Comment:
> What happens if there are multiple scripts that reference a package that doesn't exist. Do we report all of them separately?

We report each separately, yeah.

I'd push back on that error message. Like, when writing TypeScript, what I want to get is:

```
Cannot find module './bar' or its corresponding type declarations.
    import {foo} from './bar';
                       ~~~~~
```

Rather than:

```
Could not find bar.ts in <dir>

We looked for it because the following scripts reference it:
    foo.ts
```

_sometimes_ it's the case that a file is missing on disk for some reason and I have a bunch of import resolution errors, which is a little annoying, but more often it's a misspelled import specifier, and it's really nice to have that pointed out

Reply:
Good point, agreed.